geopandas/geodataframe.py

```diff
@@ -1333,6 +1333,11 @@ def explode(self):
         1 0  name2  POINT (2.00000 1.00000)
           1  name2  POINT (0.00000 0.00000)
         """
+
+        if column != self.geometry.name:
```

Comment:
This should check for dtype of a column (see the main comment).

Reply:
Also, if no `column` is specified, default to active geometry to keep backwards compatibility.